source/solvers/postprocessing_cfd.cc

```diff
@@ -159,6 +159,151 @@ calculate_pressure_drop<3, TrilinosWrappers::MPI::BlockVector>(
   const unsigned int                        inlet_boundary_id,
   const unsigned int                        outlet_boundary_id);
 
+template <int dim, typename VectorType>
+double
+calculate_momentum_drop(const DoFHandler<dim> &       dof_handler,
+                        std::shared_ptr<Mapping<dim>> mapping,
+                        const VectorType &            evaluation_point,
+                        const Quadrature<dim - 1> &   face_quadrature_formula,
+                        const unsigned int            inlet_boundary_id,
+                        const unsigned int            outlet_boundary_id,
+                        PhysicalPropertiesManager &   properties_manager)
+{
+  const FESystem<dim, dim> fe = dof_handler.get_fe();
+
+  const unsigned int               n_q_points = face_quadrature_formula.size();
+  const FEValuesExtractors::Vector velocities(0);
+  const FEValuesExtractors::Scalar pressure(dim);
+  std::vector<Tensor<1, dim>>      velocity_values(n_q_points);
+  std::vector<double>              pressure_values(n_q_points);
+  Tensor<1, dim>                   normal_vector;
+
+  FEFaceValues<dim> fe_face_values(*mapping,
+                                   fe,
+                                   face_quadrature_formula,
+                                   update_values | update_quadrature_points |
+                                     update_JxW_values | update_normal_vectors);
+
+  auto                    density_model = properties_manager.get_density(0);
+  std::map<field, double> field_values;
+  field_values[field::temperature] = 0;
+  double fluid_density             = density_model->value(field_values);
+
+  double momentum_outlet_boundary = 0;
+  double outlet_surface           = 0;
+  double momentum_inlet_boundary  = 0;
+  double inlet_surface            = 0;
+  double momentum_drop            = 0;
+
+  // Calculating area and volumetric flow rate at the inlet flow
+  for (const auto &cell : dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned() && cell->at_boundary())
+        {
+          for (unsigned int face = 0; face < GeometryInfo<dim>::faces_per_cell;
+               face++)
+            {
+              if (cell->face(face)->at_boundary())
+                {
+                  if (cell->face(face)->boundary_id() == inlet_boundary_id)
```

Comment:
I feel there's a lot of code repetition in what follows (all the get functions values, etc.). Why not make this in a single block and then do the addition in a if depending on inlet or outlet? Would make the code easier to read.